src/Discord.Net.Core/Entities/Guilds/IGuild.cs

```diff
@@ -83,10 +85,12 @@ public interface IGuild : IDeletable, ISnowflakeEntity
         /// <summary>
         ///     Gets the URL of this guild's splash image.
         /// </summary>
+        /// <param name="format">The format to return. Mustn't be a gif.</param>
+        /// <param name="size">The size of the image to return in. This can be any power of two between 16 and 2048.</param>
         /// <returns>
         ///     A URL pointing to the guild's splash image; <c>null</c> if none is set.
         /// </returns>
-        string SplashUrl { get; }
+        string GetSplashUrl(ImageFormat format = ImageFormat.Auto, ushort size = 128);
```

Comment:
Would probably be better as an extension method

Reply:
Agreed. Preserve the property definitions to what they're already doing now, and just expose the method publically for people that want to explicitly control what they get back.

Additionally, reverse the parameter order so that it's consistent with already existing methods and we don't need to break those.

Reply:
Alright, noted. Should I keep both the properties and the methods or reverse the method changes and just let them be in the CDN file? Methods are already public there.